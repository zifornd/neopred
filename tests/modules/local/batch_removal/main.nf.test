nextflow_process {

    name "Test Process BATCH_REMOVAL"
    script "modules/local/batch_removal/main.nf"
    process "BATCH_REMOVAL"

    test("Should run without failures") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            process {
                """
                // define inputs of the process here. Example:
                input[0] = file("${projectDir}/assets/samplesheet.csv")
                input[1] = params.batch
                input[2] = params.design
                input[3] = file("${projectDir}/tests/test_result/salmon.merged.gene_tpm.tsv")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
