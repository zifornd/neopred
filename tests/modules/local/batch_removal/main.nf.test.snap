{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "design_batch_tpm.genesymbol.batchremoved.csv:md5,da2e75de4c4a97c2c7db1c3e73170518"
                ],
                "1": [
                    "design_batch_tpm.genesymbol.csv:md5,9010dbd7a39a05307f67be4ab4c89bdd"
                ],
                "2": [
                    "versions.yml:md5,23137a65b62b5ab3c80949edb8f8ddb5"
                ],
                "after_br": [
                    "design_batch_tpm.genesymbol.batchremoved.csv:md5,da2e75de4c4a97c2c7db1c3e73170518"
                ],
                "before_br": [
                    "design_batch_tpm.genesymbol.csv:md5,9010dbd7a39a05307f67be4ab4c89bdd"
                ],
                "versions": [
                    "versions.yml:md5,23137a65b62b5ab3c80949edb8f8ddb5"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.1"
        },
        "timestamp": "2025-01-02T12:54:38.437775"
    }
}